src/JBRulesets.sol

```diff
@@ -952,27 +953,27 @@ contract JBRulesets is JBControlled, IJBRulesets {
         // slither-disable-next-line incorrect-equality
         if (rulesetId == 0) return ruleset;
 
-        ruleset.id = rulesetId;
+        ruleset.id = uint48(rulesetId);
 
         uint256 packedIntrinsicProperties = _packedIntrinsicPropertiesOf[projectId][rulesetId];
 
-        // `weight` in bits 0-87 bits.
-        ruleset.weight = uint256(uint88(packedIntrinsicProperties));
-        // `basedOnId` in bits 88-143 bits.
-        ruleset.basedOnId = uint256(uint56(packedIntrinsicProperties >> 88));
-        // `start` in bits 144-199 bits.
-        ruleset.start = uint256(uint56(packedIntrinsicProperties >> 144));
-        // `cycleNumber` in bits 200-255 bits.
-        ruleset.cycleNumber = uint256(uint56(packedIntrinsicProperties >> 200));
+        // `weight` in bits 0-111 bits.
+        ruleset.weight = uint112(packedIntrinsicProperties);
+        // `basedOnId` in bits 112-159 bits.
+        ruleset.basedOnId = uint48(packedIntrinsicProperties >> 112);
```

Comment:
Should we have a lib for all of these pointer indexes?

Reply:
ah, the indexes aren't relevant to the outside world, it's just packing details.